### PR TITLE
build: Add bootstrap.sh for hiba

### PR DESCRIPTION
**Pull request by wltu**
This script is needed for bitbake to build it properly in
https://gbmc-review.googlesource.com/c/meta-gbmc-staging/+/2001

Tested:
Able to build hiba after running bootstrap.sh
```
./bootstrap.sh
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I m4
autoreconf: running: /usr/bin/autoconf --force
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:13: installing './compile'
configure.ac:17: installing './missing'
autoreconf: Leaving directory '.'
Run "./configure ${CONFIGURE_FLAGS} && make"
```

Signed-off-by: Willy Tu <wltu@google.com>

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+AUTOCONF_FILES="Makefile.in aclocal.m4 ar-lib autom4te.cache compile \
+        config.guess config.h.in config.sub configure depcomp install-sh \
+        ltmain.sh missing *libtool test-driver"
+
+case $1 in
```
Comment (blunderer) [inline bootstrap.sh:7]: Without an argument, this basically does the equivalent of ./autogen.sh which will be confusion for users.

File: bootstrap.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+AUTOCONF_FILES="Makefile.in aclocal.m4 ar-lib autom4te.cache compile \
+        config.guess config.h.in config.sub configure depcomp install-sh \
+        ltmain.sh missing *libtool test-driver"
```
Comment (blunderer) [inline bootstrap.sh:5]: This is the list of files created by autotools. Can you expand on what fails if these files are already present?

File: bootstrap.sh (hunk 1 above)
Comment (blunderer) [reply bootstrap.sh:7]: We could, but I am still trying to understand what this does and why this is need

File: bootstrap.sh (hunk 2 above)
Comment (wltu) [reply bootstrap.sh]: I think it nothing will fail if it is not cleaned up before running the script again. This is just for cleanup after if we want to.

File: bootstrap.sh (hunk 1 above)
Comment (wltu) [reply bootstrap.sh:7]: Seems like bootstrap.sh is prefered over autogen.sh in https://www.sourceware.org/autobook/autobook/autobook_43.html

As for where/how this is called in bitbake... still trying to find it.

**Comment by wltu** [comment]
Tested again and it seems like this is not really needed.

It seems like I was setting up the bitbake wrong. Will close this.

**Pull request closed without merging**